src/main/java/com/gregtechceu/gtceu/forge/ForgeCommonEventListener.java

```diff
@@ -371,6 +376,85 @@ public static void onEntityLivingFallEvent(LivingFallEvent event) {
         }
     }
 
+    public static final UUID BLOCK_SPEED_BOOST = UUID.fromString("b14c1720-b06f-40f6-98fd-625af4ed1076");
+
+    @SubscribeEvent
+    public static void playerTickEvent(TickEvent.PlayerTickEvent event) {
+        Player player = event.player;
+        if (event.phase == TickEvent.Phase.START && !player.level().isClientSide) {
+            var speedAttrib = player.getAttribute(Attributes.MOVEMENT_SPEED);
+            if (speedAttrib == null) return;
+            var speedMod = speedAttrib.getModifier(BLOCK_SPEED_BOOST);
+
+            float speedBoost = 0.0f;
+            if (!player.onGround() || player.isInWater() || player.isCrouching()) {
+                speedBoost = 0.0f;
+            } else {
+                var state = player.level().getBlockState(player.getOnPos());
+                if (state.is(CustomTags.SPEED_CONCRETES)) {
+                    speedBoost = (float) speedAttrib.getBaseValue() * 0.6f; // value that is added to the base MC speed
+                } else if (state.is(CustomTags.STUDS)) {
+                    speedBoost = (float) speedAttrib.getBaseValue() * 0.25f; // slower to walk on studs
+                } else if (state.is(CustomTags.SLOW_FRAMES)) {
+                    speedBoost = -0.025f; // slower on frames
+                }
+            }
+            if (speedMod != null) {
+                if (speedBoost == speedMod.getAmount())
+                    return;
+                else
+                    speedAttrib.removeModifier(BLOCK_SPEED_BOOST);
+            } else {
+                if (speedBoost == 0.0f) return;
+            }
+            if (speedBoost != 0.0f) {
+                speedAttrib.addTransientModifier(new AttributeModifier(BLOCK_SPEED_BOOST, "GT Block Speed Boost",
+                        speedBoost, AttributeModifier.Operation.ADDITION));
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public static void updateFOV(ComputeFovModifierEvent event) {
+        Player player = event.getPlayer();
+
+        AttributeInstance moveSpeed = player.getAttribute(Attributes.MOVEMENT_SPEED);
+        if (moveSpeed == null || moveSpeed.getModifier(BLOCK_SPEED_BOOST) == null) return;
+        float originalFov = player.getAbilities().flying ? 1.1F : 1.0F;
```

Comment:
You should use https://mcstreetguy.github.io/ForgeJavaDocs/1.20.1-latest/net/minecraftforge/client/event/ComputeFovModifierEvent.html's fields (etiher getFovModifier() or getNewFovModifier()) to allow interaction with other mods